lib/puppet/pops/types/type_factory.rb

```diff
@@ -40,6 +40,13 @@ def self.float
     PFloatType::DEFAULT
   end
 
+  # Produces the Sensitive type
+  # @api public
+  #
+  def self.sensitive
+    PSensitiveType::DEFAULT
+  end
```

Comment:
You want to be able to create parameterized types, so something like:
```
def self.sensitive(t = PAnyType)
  PSensitiveType.new(t)
end
```
Many of the other types use a `nil` element_type (or equivalent) to mean `Any` as that makes it easier to share the singleton `DEFAULT`, but it is not required to do so (it requires a bit more logic in a couple of other places if @element_type.nil? should mean the same as @element_type = PAnyType).